compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/FunctionReference.kt

```diff
@@ -59,11 +59,42 @@ public sealed class FunctionReference {
       message = "Unable to get the return type for function $fqName."
     )
 
-  public abstract fun resolveGenericReturnTypeOrNull(
+  protected abstract fun resolveGenericReturnTypeOrNull(
     implementingClass: ClassReference.Psi
   ): ClassReference?
 
-  public fun resolveGenericReturnType(implementingClass: ClassReference.Psi): ClassReference =
+  public fun resolveGenericReturnTypeOrNull(
+    implementingClass: ClassReference
+  ): ClassReference? {
```

Comment:
I found the relationship between this and the protected version a bit confusing; it might help to give the protected method a different name just for easier differentiation on which one is being used where, even if it's just `resolveGenericReturnTypeOrNullInternal`, `resolveGenericReturnTypeOrNullForPsi` or `_resolveGenericReturnTypeOrNull`

Reply:
Fair point. I've changed it to `resolveGenericReturnTypeOrNullPsi`.